src/main/java/com/microsoft/graph/serializer/EdmNativeTypeSerializer.java

```diff
@@ -0,0 +1,60 @@
+package com.microsoft.graph.serializer;
+
+import java.lang.reflect.Type;
+import java.math.BigDecimal;
+import java.util.UUID;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import com.microsoft.graph.logger.ILogger;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/** Deserializer for native EDM types from the service. Used for actions and functions that return native types for example. */
+public class EdmNativeTypeSerializer {
+    /**
+     * Deserializes native EDM types from the service. Used for actions and functions that return native types for example.
+     * @param <T> Expected return type.
+     * @param json json to deserialize.
+     * @param type class of the expected return type.
+     * @param logger logger to use.
+     * @return the deserialized type or null.
+     */
+    @Nullable
+    public static <T> T deserialize(@Nonnull final JsonElement json, @Nonnull final Class<T> type, @Nonnull final ILogger logger) {
+        if (json == null || type == null) {
+			return null;
+        } else if(json.isJsonPrimitive()) {
+            return getPrimitiveValue(json, type);
+		} else if(json.isJsonObject()) {
```

Comment:
nit: indent

Reply:
those damn tabs keep creeping on me. Thanks for noticing.

Reply:
GitHub makes it easy to show these because of weird tab width choice, whether it is a good or a bad thing.

I am really curious to know as to why VSCode is not applying .editorconfig on save for you.

Reply:
it'd be interesting to see if using VS code on this repo (just check-out this branch to the commit before I fixed that mixed indent style), hitting ctrl + S on the file, it fixes it automatically for you. I had to do an explicit ctrl + h.
Here is the extension I'm using https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig

Reply:
let me give it a try.

Reply:
It turns out space conversion is not available `on save`: https://github.com/editorconfig/editorconfig-vscode#supported-properties.

If all your projects are using spaces, you may want to configure `editor.insertSpaces` in vscode settings. `editor.renderWhitespace` also helps in visualizing space types, so that you can tell when you need to hit that extra ctrl+h.

There is also this extension, but it ignores the space configuration in .editorconfig. It always applies 4 spaces: https://marketplace.visualstudio.com/items?itemName=redlin.remove-tabs-on-save

Visual Studio applies the .editorconfig spacing rules on save for C# files, so my expectation was similar with VSCode.

IntellijIDEA doesn't apply tab->space on save though it inserts new line at the end. But if you format the code, it does the conversion and respects the .editorconfig rules (tested with 8 spaces).

Maybe a little friction on Java development in VSCode to "share" in that BarcelonaJUG conference ;)

Reply:
An alternative to Ctrl+H for tabs to spaces is: Command Palette (F1 or Ctrl+Shift+P) -> Convert Indentation to Spaces, which respects `.editorconfig`.

Reply:
Thanks for looking into it! This is probably another case of Vincent getting confused by the tools. I do have auto final line insertion on save in TypeScript and yaml but this is probably because of other extensions. I do also have white space trim on save for all languages I believe?
I'll try to make the ctrl shift p alternative a habit and maybe if I get tired of doing that I'll go pr the editor config extension.